apps/accounts/tests/test_admin.py

```diff
@@ -360,60 +373,98 @@ def test_archived_providers_hidden_by_seen_with_override_params(
 
         hosting_provider_with_sample_user.archived = archived[0]
         hosting_provider_with_sample_user.save()
-        client.force_login(hosting_provider_with_sample_user.user_set.first())
+        client.force_login(hosting_provider_with_sample_user.users.first())
 
         admin_url = urls.reverse("greenweb_admin:accounts_hostingprovider_changelist")
         resp = client.get(admin_url, {"archived": True}, follow=True)
 
         assert len(resp.context["results"]) == archived[1]
         assert resp.status_code == 200
 
-    def test_staff_can_remove_provider_from_user(
-        self, db, client, hosting_provider_with_sample_user, greenweb_staff_user
+    def test_list_of_users_who_can_manage_provider_is_displayed(
+        self, db, sample_hoster_user, greenweb_staff_user, hosting_provider
     ):
-        """
-        Can a staff member update a user to remove them
-        from a given hosting provider?
-        """
+        hosting_provider.save()
+        # given: 2 distinct users have explicit permissions to manage the datacenter
+        assign_perm(str(manage_provider), sample_hoster_user, hosting_provider)
+        assign_perm(str(manage_provider), greenweb_staff_user, hosting_provider)
 
-        user = hosting_provider_with_sample_user.user_set.first()
-        provider = hosting_provider_with_sample_user
-        # log the staff member in
-        client.force_login(greenweb_staff_user)
+        # when the provider admin page is accessed by one of the users
+        provider_admin = ac_admin.HostingAdmin(
+            ac_models.Hostingprovider, admin_site.greenweb_admin
+        )
 
-        user_admin_url = urls.reverse(
-            "greenweb_admin:accounts_user_change",
-            args=[user.id],
+        # Then: they should see a list of 2 users listed on that page
+        assert sample_hoster_user.username in provider_admin.managed_by(
+            hosting_provider
+        )
+        assert greenweb_staff_user.username in provider_admin.managed_by(
+            hosting_provider
         )
 
-        user_payload_without_provider_allocated = {
-            "username": user.username,
-            "email": user.email,
-            "is_active": "on",
-            "is_staff": "on",
-            # simulate sending the empty result for a cleared
-            # hosting provider
-            "hostingprovider": "",
-            "groups": [group.id for group in user.groups.all()],
-        }
+    def test_admins_can_access_permissions_management(
+        self, db, greenweb_staff_user, hosting_provider, sample_hoster_user
+    ):
+        hosting_provider.save()
+        # when the provider admin page is accessed by admin user
+        provider_admin = ac_admin.HostingAdmin(
+            ac_models.Hostingprovider, admin_site.greenweb_admin
+        )
+        request = MagicMock()
```

Comment:
We've previously used [django.test.RequestFactory](https://docs.djangoproject.com/en/4.1/topics/testing/advanced/#django.test.RequestFactory) to create requests for using in testing like this, as it's well documented in the django docs. 

Is there specific reason for using MagicMock? If not, would you mind using the django docs? The docs are slightly easier to read, and finding good docs about mocks can be a pain.



If not, would you mind using the RequestFactory?

Reply:
I this specific test case, I need to pass any object to the view function that will have a `user` attribute set correctly, and the behavior I'm testing is only depending on this parameter - that's why I used `MagicMock`. It would be inconvenient to use `RequestFactory` in that specific test case, because I'm covering 3 different views